pkg/api/marshal_test.go

```diff
@@ -0,0 +1,334 @@
+package api
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/promql"
+)
+
+func TestMarshalVector(t *testing.T) {
+	testCases := []struct {
+		name     string
+		value    promql.Vector
+		warnings []string
+		expected string
+	}{
+		{
+			name:     "nil",
+			value:    promql.Vector{},
+			expected: `{"status":"success","data":{"resultType":"vector","result":[]}}`,
+		},
+		{
+			name:     "empty",
+			value:    promql.Vector{},
+			expected: `{"status":"success","data":{"resultType":"vector","result":[]}}`,
```

Comment:
hmm do we need these expected strings? Can't we just use the standard json encoder for the source of truth?

Reply:
the standard json encoder vs string is a bit of a tradeoff, on the one hand the standard encoder guarantees that we’ll always match the output prom expects, on the other hand it’s nice having the strings for documentation purposes…